src/app/talent/talent/talent.component.html

```diff
@@ -88,12 +88,12 @@
 						<span>Type</span>
 					</div>
 				</div>
-				<div class="row" *ngFor="let media of medias">
+				<div class="row" *ngFor="let media of medias; let i = index">
 					<div class="cell">
 						<span>{{media?.title}}</span>
 					</div>
 					<div class="cell">
-						<input class="checkbox" type="checkbox" name="published" [(ngModel)]="media.published" [ngModelOptions]="{standalone: true}">
+						<input class="checkbox" type="checkbox" name="published" (click)="togglePublished(i)" [(ngModel)]="media.published" [ngModelOptions]="{standalone: true}">
```

Comment:
instead of sending the index of the media in the array, you can send the media item directly
togglePlublished(media)